test/export/InstanceExporter.test.ts

```diff
@@ -916,6 +916,51 @@ describe('InstanceExporter', () => {
       });
     });
 
+    it('should assign elements with soft indexing used within a path', () => {
+      const assignedValRule = new AssignmentRule('name[+].given');
+      assignedValRule.value = 'John';
+      patientInstance.rules.push(assignedValRule);
+      const assignedValRule2 = new AssignmentRule('name[=].family');
+      assignedValRule2.value = 'Johnson';
+      patientInstance.rules.push(assignedValRule2);
+      const assignedValRule3 = new AssignmentRule('name[+].given');
+      assignedValRule3.value = 'Johnny';
+      patientInstance.rules.push(assignedValRule3);
+      const assignedValRule4 = new AssignmentRule('name[=].family');
+      assignedValRule4.value = 'Jackson';
+      patientInstance.rules.push(assignedValRule4);
+      const exported = exportInstance(patientInstance);
+      expect(exported.name).toEqual([
+        {
+          given: ['John'],
+          family: 'Johnson'
+        },
+        {
+          given: ['Johnny'],
+          family: 'Jackson'
+        }
+      ]);
+    });
+
+    it('should assign elements with soft indexing used on multiple elements within a path', () => {
+      const assignedValRule = new AssignmentRule('name[+].given');
+      assignedValRule.value = 'John';
+      patientInstance.rules.push(assignedValRule);
+      const assignedValRule2 = new AssignmentRule('name[=].family');
+      assignedValRule2.value = 'Johnson';
+      patientInstance.rules.push(assignedValRule2);
+      const assignedValRule3 = new AssignmentRule('name[=].given[+]');
+      assignedValRule3.value = 'Johnny';
+      patientInstance.rules.push(assignedValRule3);
+      const exported = exportInstance(patientInstance);
+      expect(exported.name).toEqual([
+        {
+          given: ['John', 'Johnny'],
+          family: 'Johnson'
+        }
+      ]);
+    });
+
```

Comment:
I think it would be good to have some more tests.  Specifically:
* a test w/ rulesets (as previously noted)
* a test (or two) w/ caret rules (since there is some significant logic there)
* tests to ensure soft indexing gets applied in each required case: Profiles, Extensions, ValueSets, CodeSystems, Instance

Reply:
In addition to these, I would like to see a test that includes both hard and soft indexing, assuming such a thing is allowed.